src/goto-symex/symex_main.cpp

```diff
@@ -519,3 +525,115 @@ void goto_symext::symex_step(
     UNREACHABLE;
   }
 }
+
+/// Collect all sub-expressions of an expression that are pointer-typed symbols
+/// \param expr: The expression to examine
+/// \param [out] output: The set to put the results into
+static void
+collect_pointer_typed_symbols(const exprt &expr, std::set<symbol_exprt> &output)
+{
+  const symbol_exprt *symbol_expr = expr_try_dynamic_cast<symbol_exprt>(expr);
```

Comment:
Considering the only user is only interested in a single pointer, could avoid allocating a pointless set by making it `find_single_pointer_typed_symbol`

Reply:
I don't see a nice way of doing that which also checks that there's only one.